airflow/www/views.py

```diff
@@ -1149,11 +1149,10 @@ def tree(self):
         default_dag_run = conf.getint('webserver', 'default_dag_run_display_number')
         dag_id = request.args.get('dag_id')
         blur = conf.getboolean('webserver', 'demo_mode')
-        dag_model = DagModel.get_dagmodel(dag_id)
-        if not dag_model:
-            flash('DAG "{0}" seems to be missing in database.'.format(dag_id), "error")
+        dag = dagbag.get_dag(dag_id)
```

Comment:
We'd like to stay using the DagModel if possible. Is this change required?

Reply:
For the Graph view, no, but the tree view seems to be broken for SubDAGs

The issue seems to arise from the `dag` object being `None` when using `DagModel.get_dagmodel` for a `dag_id` that corresponds to a subDAG

We could merge this PR without the change from `DagModel`, but then if people are using `tree` as their default view the "Zoom into Sub DAG" link will result in an error

I switched to `dagbag.get_dag` because that's the method used by the `graph` view and that view works with subDAGs

Reply:
Ah right, sounds sensible.

Reply:
I switched it to "fall back" to the `DagBag` approach instead of replacing `DagModel` wholesale